library/src/main/java/bio/terra/landingzone/stairway/flight/create/resource/step/CreatePostgresqlDbStep.java

```diff
@@ -212,6 +215,25 @@ private void enablePgBouncer(String mrgName, String postgresName) {
     }
   }
 
+  /*
+   Current step is responsible for multiple postgres operations such as 1) DB server provisioning
+   2) Enabling pgbouncer 3) Creating admin user. This particular method works as a global handler for
+   the whole step. But current implementation handles specific Postgres db provisioning issue.
+  */
+  protected Optional<StepResult> maybeHandleManagementException(ManagementException e) {
+    final String resourceOperationFailure = "ResourceOperationFailure";
+    final String internalServerError = "InternalServerError";
+    if (e.getValue() != null
+        && StringUtils.equalsIgnoreCase(e.getValue().getCode(), resourceOperationFailure)
+        && e.getValue().getDetails() != null
+        && e.getValue().getDetails().stream()
+            .anyMatch(d -> StringUtils.equalsIgnoreCase(d.getCode(), internalServerError))) {
+      logger.warn("Postgres provisioning failure. Error: {}.", e.getMessage());
```

Comment:
Should this use `ManagementExceptionUtils::buildErrorInfo` as well?

Reply:
In this particular situation I think it is not necessary to use it, but we probably can.